### PR TITLE
Use service-account to join if not --use-bootstrap-token on init

### PR DESCRIPTION
**Pull request by itdove**
https://github.com/open-cluster-management-io/clusteradm/issues/16
Signed-off-by: Dominique Vernier <dvernier@redhat.com>
/hold

### CODE REVIEW

File: pkg/helpers/client.go [hunk 1]
```diff
@@ -30,7 +32,7 @@ func GetAPIServer(kubeClient kubernetes.Interface) (string, error) {
 	return cluster.Server, nil
 }
 
-func GetCACert(kubeClient kubernetes.Interface) ([]byte, error) {
+func GetCACert(kubeClient kubernetes.Interface, useBootstrapToken bool) ([]byte, error) {
```
Comment (qiujian16) [inline pkg/helpers/client.go:35]: do you need useBootstrapToken?

File: pkg/cmd/init/exec.go [hunk 2]
```diff
@@ -92,18 +120,31 @@ func (o *Options) run() error {
...
-		o.values.Hub.TokenSecret,
+		token,
 		restConfig.Host,
 	)
 
 	return apply.WriteOutput(o.outputFile, output)
 }
+
+func waitForBootstrapSecret(kubeClient kubernetes.Interface, b wait.Backoff) (secret *corev1.Secret, err error) {
+	err = retry.OnError(b, func(err error) bool {
```
Comment (qiujian16) [inline pkg/cmd/init/exec.go:139]: i might use watch with fiedselector here. But it is a blocker.

File: pkg/helpers/client.go (hunk 1 above)
Comment (itdove) [reply pkg/helpers/client.go:35]: You are correct it is not needed anymore

File: pkg/cmd/init/exec.go (hunk 2 above)
Comment (itdove) [reply pkg/cmd/init/exec.go:139]: I don't know how to use watch as it is not a controller. Do you mean it is a blocker or not?

Comment (qiujian16) [reply pkg/cmd/init/exec.go:139]: not blocker, sorry. Updated the comment

**Review by qiujian16 — COMMENTED** [review]
/lgtm

**Pull request merged**